### PR TITLE
#695: fix segfault and unicode errors for non text parameter wkt loads

### PR DESCRIPTION
**Pull request by tbicr**
fix for https://github.com/Toblerity/Shapely/issues/695

### CODE REVIEW

File: shapely/geos.py [hunk 1]
```diff
@@ -257,6 +257,8 @@ def __del__(self):
 
     def read(self, text):
         """Returns geometry from WKT"""
+        if not isinstance(text, (bytes, str) if sys.version_info[0] >= 3 else (str, unicode)):
+            raise TypeError("Only bytes or string types can be processed.")
```
Comment (sgillies) [inline shapely/geos.py]: The problem here isn't Shapely's WKT reader, it's ctypes' c_char_p, which interprets integers as pointers to memory. A `if isinstance(text, int): raise TypeError(...)` will suffice – we only need to intercept the int case. See https://bugs.python.org/issue1701409 for more details.

Comment (tbicr) [reply shapely/geos.py]: in this case there are another issue for python3 https://github.com/Toblerity/Shapely/blob/master/shapely/geos.py#L261 which will raise `AttributeError`

Comment (sgillies) [reply shapely/geos.py]: I agree 👍

Comment (tbicr) [reply shapely/geos.py]: so what the best approach for this bug fixing:
1. use byte/string instance parameter check?
2. use int parameter check + add AttributeError handling and TypeError reraising?

Comment (sgillies) [reply shapely/geos.py]: I'm sorry that I wasn't clear: number 2, please.

Comment (tbicr) [reply shapely/geos.py]: one more question, for python 3 `bytes` type doesn't allowed now, should be this case covered or only unicode string should be allowed?

Comment (tbicr) [reply shapely/geos.py]: added bytes handling for python3

File: shapely/geos.py [hunk 2]
```diff
@@ -257,7 +257,9 @@ def __del__(self):
 
     def read(self, text):
         """Returns geometry from WKT"""
-        if sys.version_info[0] >= 3:
+        if isinstance(text, int):
+            raise TypeError("Only bytes or string types can be processed.")
+        if sys.version_info[0] >= 3 and isinstance(text, str):
             text = text.encode('ascii')
         geom = self._lgeos.GEOSWKTReader_read(self._reader, c_char_p(text))
```
Comment (sgillies) [inline shapely/geos.py]:
```suggestion
        c_string = c_char_p(text)
        geom = self._lgeos.GEOSWKTReader_read(self._reader, c_string)
```

File: shapely/geos.py (hunk 1 above)
Comment (tbicr) [reply shapely/geos.py]: updated

**Pull request merged**